bitsandbytes/__main__.py

```diff
@@ -11,32 +11,18 @@
 
 HEADER_WIDTH = 60
 
-def execute_and_return(command_string: str) -> Tuple[str, str]:
-    def _decode(subprocess_err_out_tuple):
-        return tuple(
-            to_decode.decode("UTF-8").strip()
-            for to_decode in subprocess_err_out_tuple
-        )
-
-    def execute_and_return_decoded_std_streams(command_string):
-        return _decode(
-            subprocess.Popen(
-                shlex.split(command_string),
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            ).communicate()
-        )
-
-    std_out, std_err = execute_and_return_decoded_std_streams(command_string)
-    return std_out, std_err
 
 def find_file_recursive(folder, filename):
-    cmd = f'find {folder} -name {filename}'
-    out, err = execute_and_return(cmd)
-    if len(err) > 0:
-        raise RuntimeError('Something when wrong when trying to find file. Maybe you do not have a linux system?')
+    import glob
+    outs = []
+    try:
+        for ext in ["so", "dll", "dylib"]:
+            out = glob.glob(os.path.join(folder, "**", filename + ext))
+            outs.extend(out)
+    except Exception as e:
+        raise RuntimeError('Error: Something when wrong when trying to find file. {e}')
 
-    return out
+    return outs
```

Comment:
Does the list instead of path cause errors in some edge cases? As I understand it, we either have `dll` or `dylib`, is that correct? Or can we have both in some cases?